tls/s2n_cipher_suites.c

```diff
@@ -1132,13 +1132,18 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t * wire,
                 match = match->sslv3_cipher_suite;
             }
 
-            /* Make sure the cipher is valid for available certs */
-            if (s2n_is_cipher_suite_valid_for_auth(conn, match) != S2N_SUCCESS) {
+            /* Skip the suite if we don't have an available implementation */
```

Comment:
are you sure that this change do not impact previous versions of TLS ?

Reply:
this code is from the existing code, i re-ordered it so we check certificates (in s2n_is_cipher_suite_valid_for_auth()) only after making sure that the ciphersuite is available to be used.